### PR TITLE
add plv8 formula

### PR DESCRIPTION
**Pull request by JerrySievert**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by JerrySievert** [comment]
it appears that the test failure is due to permissions/ownership of the extensions directory for postgres

File: Formula/plv8.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "https://plv8.github.io/"
+  url "https://github.com/plv8/plv8/archive/v2.3.3.zip"
+  sha256 "324660045f838c549b5244d8c7dea84612430b12e00bf3a8a61fb500a9954de8"
+
+  depends_on "postgresql"
+  depends_on "python@2"
+
+  def install
+    system 'pg_config' # ensure postgres installed
+    system "make"
```
Comment (fxcoudert) [inline Formula/plv8.rb]: `make` appears to be downloading v8 and building it. This needs to depend on our v8 formula, not rebuild it.

File: Formula/plv8.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    system "make install"
+  end
+
+  test do
+    ENV.prepend 'PATH', Formula.factory('postgresql').bin, ':'
```
Comment (fxcoudert) [inline Formula/plv8.rb]: `ENV.prepend_path`

**Review by fxcoudert — CHANGES_REQUESTED** [review]
```
  * C: 4: col 3: Use GitHub tarballs rather than zipballs (url is https://github.com/plv8/plv8/archive/v2.3.3.zip).
  * C: 11: col 12: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 17: col 17: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 17: col 25: "Formula.factory(name)" is deprecated in favor of "Formula[name]"
  * C: 17: col 41: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 17: col 60: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * Use `system "make", "install"` instead of `system "make install"` 
  * Use `system "make", "installcheck"` instead of `system "make installcheck"` 
```

**Comment by JerrySievert** [comment]
forced revert back to audited version.

please see my note re: v8, and why the Homebrew package will not work.

File: Formula/plv8.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Plv8 < Formula
+  desc " V8 Engine Javascript Procedural Language add-on for PostgreSQL"
```
[the remark this answers is not in the document]
Comment (JerrySievert) [reply Formula/plv8.rb]: fixed due to audited version revert

**Comment by JerrySievert** [comment]
any updates on this?

**Comment by JerrySievert** [comment]
the v8 team has specifically stated that they no longer support building a shared library, and that it should not be built as one, nor is it supported in the `ninja` build system.

see: https://twitter.com/bmeurer/status/960584003148505088

I'm confused as to why this is an issue with `plv8`, which requires `v8` to be built, not able to use the Homebrew provided `v8`, but not an issue with `node`, which requires `v8` to be built, not able to use the Homebrew provided `v8`?

**Comment by JerrySievert** [comment]
I can package the `v8` source as part of the `plv8` source, but would not want to do that until a `3.0` release.

**Comment by benjie** [comment]
Not stale.

**Comment by benjie** [comment]
😔

**Comment by JerrySievert** [comment]
yup, I was wondering about that.

in the mean time, `plv8` 3.0.0 is under active development against `v8` 6.8, which isn't backward compatible with 6.4, but has `BigInt` support.

still wondering how `node` can bundle, but this can't.

**Comment by benjie** [comment]
Well, keep up the good work - it’s really easy to install PLv8 from your repo now so the need for brew is diminished 👍

**Pull request closed without merging**